OpenEdXMobile/src/main/java/org/edx/mobile/util/EventObserver.kt

```diff
@@ -0,0 +1,18 @@
+package org.edx.mobile.util
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+
+/**
+ * A non-nullable observer for [Event] class that returns the value of the event if it is not yet
+ * consumed.
+ */
+class EventObserver<T>(private val consumer: (T) -> Unit) : Observer<Event<T>> {
```

Comment:
please mention the inspiration ref and the use case to define `EventObserver`, and `Event` classes.

Reply:
android already introduced `MutableStateFlow` so better to use this instated `Event`, 

ref: https://medium.com/geekculture/atomic-updates-with-mutablestateflow-dc0331724405

Reply:
Removed the `Event` and `EventObserver` as a result of our offline discussion.